applications/teletype/src/templates/teletype_customer_update.erl

```diff
@@ -0,0 +1,157 @@
+
+-define(TEMPLATE_TEXT, <<"Dear {{user.first_name}} {{user.last_name}}.\n\nHere are some news that we have selected for you.\n\nBest regards,">>).
+-define(TEMPLATE_HTML, <<"<p>Dear {{user.first_name}} {{user.last_name}}.</p><p>Here are some news that we have selected for you.</p><p>Best regards,</p>">>).
+-define(TEMPLATE_SUBJECT, <<"Customer update">>).
+-define(TEMPLATE_CATEGORY, <<"user">>).
+-define(TEMPLATE_NAME, <<"Customer update">>).
+
+-define(TEMPLATE_TO, ?CONFIGURED_EMAILS(?EMAIL_ORIGINAL)).
+-define(TEMPLATE_FROM, teletype_util:default_from_address(?MOD_CONFIG_CAT)).
+-define(TEMPLATE_CC, ?CONFIGURED_EMAILS(?EMAIL_SPECIFIED, [])).
+-define(TEMPLATE_BCC, ?CONFIGURED_EMAILS(?EMAIL_SPECIFIED, [])).
+-define(TEMPLATE_REPLY_TO, teletype_util:default_reply_to(?MOD_CONFIG_CAT)).
+
+-spec init() -> 'ok'.
+init() ->
+    wh_util:put_callid(?MODULE),
+    teletype_templates:init(?TEMPLATE_ID, [{'macros', ?TEMPLATE_MACROS}
+                                           ,{'text', ?TEMPLATE_TEXT}
+                                           ,{'html', ?TEMPLATE_HTML}
+                                           ,{'subject', ?TEMPLATE_SUBJECT}
+                                           ,{'category', ?TEMPLATE_CATEGORY}
+                                           ,{'friendly_name', ?TEMPLATE_NAME}
+                                           ,{'to', ?TEMPLATE_TO}
+                                           ,{'from', ?TEMPLATE_FROM}
+                                           ,{'cc', ?TEMPLATE_CC}
+                                           ,{'bcc', ?TEMPLATE_BCC}
+                                           ,{'reply_to', ?TEMPLATE_REPLY_TO}
+                                          ]).
+
+-spec handle_req(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_req(JObj, _Props) ->
+    'true' = wapi_notifications:customer_update_v(JObj),
+    DataJObj = wh_json:normalize(JObj),
+    AccountId = wh_json:get_value(<<"account_id">>, DataJObj),
+    case teletype_util:is_notice_enabled(AccountId, JObj, ?TEMPLATE_ID) of
+        'false' -> lager:debug("notification handling not configured for this account");
+        'true' -> process_req(DataJObj)
+    end.
+
+-spec process_req(wh_json:object()) -> 'ok'.
+process_req(DataJObj) ->
+    SenderId = wh_json:get_value(<<"account_id">>, DataJObj),
+    ViewOpts = [{'startkey', [SenderId]}
+               ,{'endkey', [SenderId, wh_json:new()]}
+               ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, <<"accounts/listing_by_children">>, ViewOpts) of
+        {'ok', Accounts} ->
+            process_accounts(Accounts, DataJObj);
+        {'error', _Reason} = E ->
+            lager:info("failed to load children. error: ~p", [E]),
+            E
+    end.
+
+process_accounts(Accounts, DataJObj) ->
+    case wh_json:get_value(<<"recipient_id">>, DataJObj) of
+        'undefined' ->
+            [process_account(wh_json:get_value(<<"value">>, Account), DataJObj) || Account <- Accounts];
+        RecipientId ->
```

Comment:
I think you should check for `recipient_id` first, and if undefined, load descendant accounts to process. Why traverse the list of accounts when you know exactly which account you want to process?